CARv2 MultiIndex.md

```diff
@@ -0,0 +1,40 @@
+# CARv2 MultiIndex
+
+![wip](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square)
+
+## Editors
+
+* [Alan Shaw](https://github.com/alanshaw), [DAG House](https://dag.house/)
+
+## Authors
+
+* [Alan Shaw](https://github.com/alanshaw), [DAG House](https://dag.house/)
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Abstract
+
+A multi-index is a CARv2 index format that allows multiple CARv2 indexes to be grouped together in a single index.
```

Comment:
I may be missing something and/or this might grow on me, but there's a pedant inside me saying that usually when we talk about multi* things in the IPFS ecosystem (ie, multihash, multiaddress) we're usually talking about a string of bytes that may be one of a number of different formats, and so when I glanced at this the first thing I thought was "oh this is about a thing that lets you disambiguate different strings of bytes that represent encoded indices" rather than "this is a thing that concatenates different indices together."

Not remotely a blocker, but might be worth clarifying or trying to find a different name - at the very least I just want to confirm that this isn't a multi* thing in the sense of multihashes or multiaddresses, is that right?

Reply:
Bonus points if you can work a Voltron or Power Rangers reference into the name since this is sort of like a Megazord for indices? 
![image](https://github.com/web3-storage/specs/assets/1113/597fa77e-846b-4e13-817f-334f16a6cd1d)

Reply:
> Not remotely a blocker, but might be worth clarifying or trying to find a different name - at the very least I just want to confirm that this isn't a multi* thing in the sense of multihashes or multiaddresses, is that right?

Yes that is correct - I'm open to a better name suggestion 🙂

`CombinedIndex`, `AggregateIndex`...? ...not sure. Want to express that it's not just a joining, since it also carries information about what CAR each index pertains to.

Reply:
This is a good feedback, I also thought about the naming creating some confusion. Could `DagIndex` or would we want to use this out of the context of representing a Dag?  I think that naming could work well given it can be an index for either a root CID having all its blocks, as well as sub-dags that could be optimized.

Reply:
@travis  I could get behind `ZordIndex`...

Reply:
* `CombinedIndex` -  join or merge to form a single unit or substance
* `CompositeIndex` - made up of several parts or elements
* `CompoundIndex` - a thing that is composed of two or more separate elements

Would be good to get a name sorted - I'd like to merge this as we're using it in production for rollup indexes in freeway.